diff -rupN openrc/mk/scripts.mk openrc-new/mk/scripts.mk
--- openrc/mk/scripts.mk	2014-01-24 12:11:40.462610384 +0100
+++ openrc-new/mk/scripts.mk	2014-01-24 12:12:37.837893095 +0100
@@ -25,8 +25,8 @@ all: ${OBJS} ${TARGETS}
 
 realinstall: ${BIN} ${CONF} ${INC}
 	@if test -n "${DIR}"; then \
-		${ECHO} ${INSTALL} -d ${DESTDIR}/${DIR}; \
-		${INSTALL} -d ${DESTDIR}/${DIR} || exit $$?; \
+		${ECHO} mkdir -p ${DESTDIR}/${DIR}; \
+		mkdir -p ${DESTDIR}/${DIR} || exit $$?; \
 	fi
 	@if test -n "${BIN}"; then \
 		${ECHO} ${INSTALL} -m ${BINMODE} ${BIN} ${DESTDIR}/${DIR}; \
